src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1377,91 +1376,61 @@ async def import_pool(self, job, data):
         """
-
-        pool = None
-        for p in await self.middleware.call('zfs.pool.find_import'):
-            if p['guid'] == data['guid']:
-                pool = p
-                break
-        if pool is None:
-            raise CallError(f'Pool with guid "{data["guid"]}" not found', errno.ENOENT)
-
+        guid = data['guid']
+        new_name = data.get('name')
+
+        # validate
+        imported_pools = await self.middleware.call('zfs.pool.query_imported_fast')
+        if guid in imported_pools:
+            raise CallError(f'Pool with guid: "{guid}" already imported', errno.EEXIST)
+        elif new_name and new_name in imported_pools.values():
+            err = f'Cannot import pool using new name: "{new_name}" because a pool is already imported with that name'
+            raise CallError(err, errno.EEXIST)
+
+        # import zpool
         try:
-            job.check_pipe("input")
-            key = job.pipes.input.r
-        except ValueError:
-            key = None
-
-        passfile = None
-        if key and data.get('passphrase'):
-            encrypt = 2
-            passfile = tempfile.mktemp(dir='/tmp/')
-            with open(passfile, 'w') as f:
-                os.chmod(passfile, 0o600)
-                f.write(data['passphrase'])
-        elif key:
-            encrypt = 1
-        else:
-            encrypt = 0
-
-        pool_name = data.get('name') or pool['name']
-        pool_id = None
-        try:
-            pool_id = await self.middleware.call('datastore.insert', 'storage.volume', {
-                'vol_name': pool_name,
-                'vol_encrypt': encrypt,
-                'vol_guid': data['guid'],
-                'vol_encryptkey': str(uuid.uuid4()) if encrypt else '',
-            })
-            pool = await self.middleware.call('pool.query', [('id', '=', pool_id)], {'get': True})
-            if encrypt > 0:
-                if not os.path.exists(GELI_KEYPATH):
-                    os.mkdir(GELI_KEYPATH)
-                with open(pool['encryptkey_path'], 'wb') as f:
-                    f.write(key.read())
-
-            await self.middleware.call('pool.scrub.create', {
-                'pool': pool_id,
-            })
-
-            await self.middleware.call('zfs.pool.import_pool', pool['guid'], {
-                'altroot': '/mnt',
-                'cachefile': ZPOOL_CACHE_FILE,
-            }, True, None, pool_name)
-
-            await self.middleware.call('zfs.dataset.update', pool_name, {
-                'properties': {
-                    'aclinherit': {'value': 'passthrough'},
-                },
-            })
+            opts = {'altroot': '/mnt', 'cachefile': ZPOOL_CACHE_FILE}
+            any_host = True
+            use_cachefile = None
+            await self.middleware.call('zfs.pool.import_pool', guid, opts, any_host, use_cachefile, new_name)
+        except Exception as e:
+            raise CallError(f'Failed importing pool with guid "{guid}" with error {e}')
```

Comment:
I think we should raise `CallError` if import pool fails in zfs plugin instead of raising `ZFSException` as that seems to be the only usage where this is not happening. In that case, we don't need to handle the exception at all and we would/should have correct codes set as well like if the pool was not available for import, a call error would already be raised with the correct error code.